src/repr/src/scalar.rs

```diff
@@ -1549,7 +1549,15 @@ impl<'a> ScalarType {
                     custom_oid: oid_b,
                     custom_name: name_b,
                 },
-            ) => fields_a.eq(fields_b) && oid_a == oid_b && name_a == name_b,
+            ) => {
+                oid_a == oid_b
+                    && name_a == name_b
```

Comment:
Think this also needs `&& fields_a.len() == fields_b.len()`

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=89b03260b03c8a1ae0bd8943d2270dbb